pkg/build/webhook/controller.go

```diff
@@ -20,7 +20,7 @@ type Plugin interface {
 	// - newly created build object or nil if default is to be created
 	// - information whether to trigger the build itself
 	// - eventual error.
-	Extract(buildCfg *buildapi.BuildConfig, secret, path string, req *http.Request) (*buildapi.SourceRevision, bool, error)
+	Extract(buildCfg *buildapi.BuildConfig, secret, path string, req *http.Request) (*buildapi.SourceRevision, []kapi.EnvVar, bool, error)
```

Comment:
so while this is technically "fine" it set off my spider sense that you were making the exact same changes in two files.  A little investigation reveals that this file is longer actually used at runtime (though we still have test code that exercises it).

Even worse, the file that replaced it (buildconfig/webhook.go) is not being exercised by those tests.

So we need a follow up to clean up this mess.